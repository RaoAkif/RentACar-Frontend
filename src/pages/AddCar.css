.add-car-main {
  margin-top: 5vh;
  display: flex;
  flex-direction: column;
  align-items: center;
}

.add-car-main > h2 {
  text-align: left;
}

#form-btn {
  margin-left: 1rem;
  background: var(--green-color);
  border: none;
  padding: 0.5rem 1.4rem;
  font-size: 1.4rem;
}

.form-control-lg .form-control {
  width: 40vw;
}