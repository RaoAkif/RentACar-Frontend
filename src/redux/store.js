import { configureStore } from '@reduxjs/toolkit';
import cakeReducer from './cake/cakeSlice';
import icecreamReducer from './icecream/icecreamSlice';
import userReducer from './user/userSlice';

const store = configureStore({
  reducer: {
    cake: cakeReducer,
    icecream: icecreamReducer,
    user: userReducer,
  },
});

export default store;
