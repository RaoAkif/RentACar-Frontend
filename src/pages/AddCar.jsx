import React from 'react';
import { useDispatch } from 'react-redux';
import { useNavigate } from 'react-router-dom';
import Button from 'react-bootstrap/Button';
import Form from 'react-bootstrap/Form';
import { addCar } from '../redux/car/car';
import './AddCar.css';

function AddCar() {
  const [car, setCar] = React.useState('');
  const dispatch = useDispatch();
  const navigate = useNavigate();

  const handleSubmitCar = (e) => {
    e.preventDefault();
    const form = e.target;
    const obj = {
      name: form.name.value,
      model: form.model.value,
      image: form.image.value,
      desc: form.desc.value,
      rent: form.rent.value,
    };
    dispatch(addCar(obj));
    navigate('/');
  };

  return (
    <div className="add-car-main">
      <h2 className="title">ADD&nbsp;&nbsp;CAR</h2>
      <Form className="add-car-form" onSubmit={handleSubmitCar}>
        <Form.Group className="form-control-lg">
          <Form.Control
            name="name"
            as="select"
            value={car}
            onChange={(e) => {
              setCar(e.target.value);
            }}
          >
            <option>Select a Car</option>
            <option value="Audi">Audi</option>
            <option value="BMW">BMW</option>
            <option value="GMC">GMC</option>
            <option value="Jeep">Jeep</option>
            <option value="Mazda">Mazda</option>
            <option value="Mercedes-Benz">Mercedes-Benz</option>
            <option value="Mitsubishi">Mitsubishi</option>
            <option value="Nissan">Nissan</option>
            <option value="Suzuki">Suzuki</option>
            <option value="Tesla">Tesla</option>
            <option value="Toyota">Toyota</option>
          </Form.Control>
        </Form.Group>
        <Form.Group className="form-control-lg">
          <Form.Control name="model" placeholder="Car Model" className="form-control-lg" />
        </Form.Group>
        <Form.Group className="form-control-lg">
          <Form.Control name="image" placeholder="Image URL" className="form-control-lg" />
        </Form.Group>
        <Form.Group className="form-control-lg">
          <Form.Control name="desc" placeholder="Write the desription..." className="form-control-lg" />
        </Form.Group>
        <Form.Group className="form-control-lg" controlId="dob">
          <Form.Control type="number" name="rent" placeholder="Rent Amount" />
        </Form.Group>
        <Button className="btn btn-white" id="form-btn" variant="primary" type="submit">
          Add Car
        </Button>
      </Form>
    </div>
  );
}

export default AddCar;
