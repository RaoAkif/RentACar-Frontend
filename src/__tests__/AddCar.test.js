import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import store from '../redux/configureStore';
import AddCar from '../pages/AddCar';

describe('AddCar page', () => {
  it('should render correctly', () => {
    const { container } = render(
      <BrowserRouter>
        <Provider store={store}>
          <AddCar />
        </Provider>
        ,
      </BrowserRouter>,
    );

    expect(container).toMatchSnapshot();
  });
});
