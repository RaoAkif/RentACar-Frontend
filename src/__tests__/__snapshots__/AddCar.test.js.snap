// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddCar page should render correctly 1`] = `
<div>
  <div
    class="add-car-main"
  >
    <h2
      class="title"
    >
      ADD  CAR
    </h2>
    <form
      class="add-car-form"
    >
      <div
        class="form-control-lg"
      >
        <input
          class="form-control-lg form-control"
          name="name"
          placeholder="Car Name"
        />
      </div>
      <div
        class="form-control-lg"
      >
        <input
          class="form-control-lg form-control"
          name="model"
          placeholder="Car Model"
        />
      </div>
      <div
        class="form-control-lg"
      >
        <input
          class="form-control-lg form-control"
          name="image"
          placeholder="Image URL"
        />
      </div>
      <div
        class="form-control-lg"
      >
        <input
          class="form-control-lg form-control"
          name="desc"
          placeholder="Write the desription..."
        />
      </div>
      <div
        class="form-control-lg"
      >
        <input
          class="form-control"
          id="dob"
          name="rent"
          placeholder="Rent Amount"
          type="number"
        />
      </div>
      <button
        class="btn btn-white btn btn-primary"
        id="form-btn"
        type="submit"
      >
        Add Car
      </button>
    </form>
  </div>
  ,
</div>
`;
